smt_switch/src/functions.py

```diff
@@ -1,480 +1,222 @@
-        if len(args) > 1:
-            return args[0].solver.apply_fun(self, *args)
-        elif len(args) == 1:
-            return args[0]
-        else:
-            return False
-
-
-class Ite(FunctionBase):
-    arity = {'min': 3,
-             'max': 3}
-
-    def __init__(self):
-        super().__init__(self.arity, '(ite cond arg1 arg2)')
-
-    def osort(self, *args):
-        if len(args) != 3:
-            raise ValueError(self.__class__name +
-                             '\'s output sort is parameterized by its arguments. ' +
-                             'Need all three inputs to determine output sort.')
-
-        arg_sorts = smtutils.sorts_list(args)
-
-        # TODO: check for consistent output sort
-        if not reduce(lambda x, y: x and y, arg_sorts[1:]):
-            raise ValueError('Ite needs to have a consistent output sort. ' +
-                             'Received {} and {}'.format(args[1].sort, args[2].sort))
-        ite_sorts = list(filter(lambda arg: arg is not None, arg_sorts[1:]))
-        if not ite_sorts:
-            raise ValueError('Can\'t determine output sort based on inputs {} and {}'.format(args[1], args[2]))
-        return ite_sorts[0]
-
-
-class Sub(FunctionBase):
-    arity = {'min': 2,
-             'max': 2}
-
-    def __init__(self):
-        super().__init__(self.arity, '(- arg1 arg2)')
-
-    def osort(self, *args):
-        if len(args) == 0:
-            raise ValueError(self.__class__name +
-                             '\'s output sort is parameterized by its arguments. ' +
-                             'Need an argument to determine output sort.')
-        return args[0].sort
-
-
-class Plus(FunctionBase):
-    arity = {'min': 2,
-             'max': __MAXARGS__}
-
-    def __init__(self):
-        super().__init__(self.arity, '(+ arg1 arg2)')
-
-    def osort(self, *args):
-        if len(args) == 0:
-            raise ValueError(self.__class__name +
-                             '\'s output sort is parameterized by its arguments. ' +
-                             'Need an argument to determine output sort.')
-        return args[0].sort
-
-
-class LT(FunctionBase):
-    arity = {'min': 2,
-             'max': 2}
-
-    def __init__(self):
-        super().__init__(self.arity, '(< arg1 arg2)')
-
-    def osort(self, *args):
-        return sorts.Bool()
-
-
-class LEQ(FunctionBase):
-    arity = {'min': 2,
-             'max': 2}
-
-    def __init__(self):
-        super().__init__(self.arity, '(<= arg1 arg2)')
-
-    def osort(self, *args):
-        return sorts.Bool()
-
-
-class GT(FunctionBase):
-    arity = {'min': 2,
-             'max': 2}
+__all__ = ['And', 'Or']  # append to list in generate functions
 
-    def __init__(self):
-        super().__init__(self.arity, '(> arg1 arg2)')
 
-    def osort(self, *args):
-        return sorts.Bool()
+fdata = namedtuple('fdata', 'num_indices, min_arity, max_arity')
+__MAXARGS__ = 6000
```

Comment:
Is there a reason why 6000? Would it make sense to use -1 for an arbitrary number of arguments?